File: system/jv/be/BECS_Runtime.java
/*
 * Copyright (c) 2015-2023, the Bennt Authors.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 *
 * Licensed under the BSD 2-Clause License (the "License").
 * See the LICENSE file in the project root for more information.
 *
 */

package be;

import java.util.*;

import be.BEC_2_6_6_SystemObject;
import be.BEC_2_5_4_LogicBool;
import be.BEC_2_6_11_SystemInitializer;

//This should be the var type to avoid unnecessary casting
//if we want to optimize "object" level calls we can put the signatures
//in this class and call directly...

//members array and mod for dynamic calls

public class BECS_Runtime {
    
    public static boolean isInitted = false;
   
    public static BEC_2_5_4_LogicBool boolTrue = new BEC_2_5_4_LogicBool(true);
    public static BEC_2_5_4_LogicBool boolFalse = new BEC_2_5_4_LogicBool(false);
    
    public static Map<String, BETS_Object> typeRefs;
    
    //for setting up initial instances
    public static BEC_2_6_11_SystemInitializer initializer;
    
    public static String platformName;
    
    public static String[] args;
    
    public static Map<String, int[]> smnlcs;
    public static Map<String, int[]> smnlecs;
    
    public static void init() { 
        if (isInitted) { return; }
        isInitted = true;
        BECS_Ids.callIds = new HashMap<String, Integer>();
        BECS_Ids.idCalls = new HashMap<Integer, String>();
        typeRefs = new HashMap<String, BETS_Object>();
        smnlcs = new HashMap<String, int[]>();
        smnlecs = new HashMap<String, int[]>();
        initializer = new BEC_2_6_11_SystemInitializer();
    }
    
    public static int getNlcForNlec(String clname, int val) {
      int[] sls = smnlcs.get(clname);
      int[] esls = smnlecs.get(clname);
      if (esls != null) {
        int eslslen = esls.length;
        for (int i = 0;i < eslslen;i++) {
          if (esls[i] == val) {
            return sls[i];
          }
        }
      } else {
        return -2; 
      }
      return -1;
    }
    
}

File: system/cs/be/BECS_Runtime.cs
// Copyright 2015 Craig Welch
//
// Licensed under the MIT license. See LICENSE.txt file in the project root 
// for full license information.

namespace be {

using System;
using System.Collections.Generic;
using be;

public class BECS_Runtime {
    
    public static bool isInitted = false;
    
    public static BEC_2_5_4_LogicBool boolTrue = new BEC_2_5_4_LogicBool(true);
    public static BEC_2_5_4_LogicBool boolFalse = new BEC_2_5_4_LogicBool(false);
    
    public static Dictionary<string, Type> typeInstances;
    
    public static Dictionary<string, BETS_Object> typeRefs;
    
    //for setting up initial instances
    public static BEC_2_6_11_SystemInitializer initializer;
    
    public static string platformName;
    
    public static string[] args;
    
    public static Dictionary<string, int[]> smnlcs;
    public static Dictionary<string, int[]> smnlecs;
    
    public static void init() { 
        if (isInitted) { return; }
        isInitted = true;
        BECS_Ids.callIds = new Dictionary<string, int>();
        BECS_Ids.idCalls = new Dictionary<int, string>();
        typeInstances = new Dictionary<string, Type>();
        typeRefs = new Dictionary<string, BETS_Object>();
        smnlcs = new Dictionary<string, int[]>();
        smnlecs = new Dictionary<string, int[]>();
        initializer = new BEC_2_6_11_SystemInitializer();
    }
    
        
    public static int getNlcForNlec(string clname, int val) {
      int[] sls = smnlcs[clname];
      int[] esls = smnlecs[clname];
      if (esls != null) {
        //Console.WriteLine("esls is not null " + clname + " val " + val);
        int eslslen = esls.Length;
        for (int i = 0;i < eslslen;i++) {
          //Console.WriteLine("esls i " + esls[i]);
          if (esls[i] == val) {
            return sls[i];
          }
        }
      } else {
        //Console.WriteLine("esls is null " + clname);
      }
      return -1;
    }
    
}

}

